File: example/src/main/java/com/github/vivchar/example/pages/github/items/StargazerViewRenderer.java
package com.github.vivchar.example.pages.github.items;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.vivchar.example.R;
import com.github.vivchar.example.widgets.CustomViewFinder;
import com.github.vivchar.rendererrecyclerviewadapter.ViewRenderer;
import com.github.vivchar.rendererrecyclerviewadapter.ViewState;

import static android.view.View.GONE;
import static android.view.View.VISIBLE;

/**
 * Created by Vivchar Vitaly on 1/10/17.
 */
public class StargazerViewRenderer extends ViewRenderer<StargazerModel, CustomViewFinder> {

    public StargazerViewRenderer(final int layoutID, @NonNull final Listener listener) {
        super(layoutID, StargazerModel.class, (model, finder, payloads) -> finder
                .setUrl(R.id.avatar, model.getAvatarUrl())
                .setOnClickListener(() -> {
                    final boolean willChecked = finder.find(R.id.check).getVisibility() == GONE;
                    finder.find(R.id.check).setVisibility(willChecked ? VISIBLE : GONE);
                    listener.onStargazerItemClicked(model, willChecked);
                })
                .setOnClickListener(R.id.check, () -> {
                    final boolean willChecked = finder.find(R.id.check).getVisibility() == GONE;
                    finder.find(R.id.check).setVisibility(willChecked ? VISIBLE : GONE);
                    listener.onStargazerItemClicked(model, willChecked);
                }));
    }

    @Nullable
    @Override
    public ViewState createViewState() {
        return new StargazerViewState();
    }

    @Override
    public int createViewStateID(@NonNull final StargazerModel model) {
        return model.getID();
    }

    public interface Listener {
        void onStargazerItemClicked(@NonNull StargazerModel model, final boolean isChecked);
    }
}

File: example/src/main/java/com/github/vivchar/example/pages/github/items/StargazerViewRenderer.kt
package com.github.vivchar.example.pages.github.items

import android.view.View
import com.github.vivchar.example.R
import com.github.vivchar.example.widgets.CustomViewFinder
import com.github.vivchar.rendererrecyclerviewadapter.BaseViewRenderer.Binder
import com.github.vivchar.rendererrecyclerviewadapter.ViewRenderer

/**
 * Created by Vivchar Vitaly on 1/10/17.
 */
class StargazerViewRenderer(layoutID: Int, listener: (StargazerModel, Boolean) -> Unit) : ViewRenderer<StargazerModel, CustomViewFinder>(
	layoutID,
	StargazerModel::class.java,
	Binder { model, finder, _ ->
		finder
			.setUrl(R.id.avatar, model.avatarUrl)
			.setOnClickListener {
				val willChecked = finder.find<View>(R.id.check).visibility == View.GONE
				finder.find<View>(R.id.check).visibility = if (willChecked) View.VISIBLE else View.GONE
				listener.invoke(model, willChecked)
			}
			.setOnClickListener(R.id.check) {
				val willChecked = finder.find<View>(R.id.check).visibility == View.GONE
				finder.find<View>(R.id.check).visibility = if (willChecked) View.VISIBLE else View.GONE
				listener.invoke(model, willChecked)
			}
	}) {

	override fun createViewState() = StargazerViewState()
	override fun createViewStateID(model: StargazerModel) = model.id
}